rust/src/delta_datafusion.rs

```diff
@@ -294,6 +304,18 @@ impl PruningStatistics for DeltaTable {
     }
 }
 
+// each delta table must register a specific object store, since paths are internally
+// handled relative to the table root.
+fn register_store(table: &DeltaTable, env: Arc<RuntimeEnv>) {
+    let object_store_url = table.storage.object_store_url();
+    let url: &Url = object_store_url.as_ref();
+    env.register_object_store(
+        url.scheme(),
+        url.host_str().unwrap_or_default(),
+        table.object_store(),
+    );
```

Comment:
IIUC, any URI passed with the same scheme and bucket name will match to this object store? And the object stores options will be preserved? I'm worried about the cases where [storage options](https://github.com/delta-io/delta-rs/blob/a2af20bd963ae7c7923d3e4030e946a6a8199779/rust/src/builder/mod.rs#L199) were directly passed into the `DeltaTable` instance.

Reply:
the object store url we get from `table.storage.object_store_url` is just for internal purposes and should not collide with common urls. It uses `delta-rs://` as a scheme and does a little bit of pruning required for widows paths.